synapse/storage/schema/main/delta/73/24_events_jump_to_date_index.sql

```diff
@@ -0,0 +1,17 @@
+/* Copyright 2023 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+INSERT INTO background_updates (ordering, update_name, progress_json) VALUES
+  (7324, 'events_jump_to_date_index', '{}');
```

Comment:
For my own curiosity, what does ordering refer to here and how did you derive the value 7324 for it?

Reply:
My entirely home-grown speculative understanding:

There can sometimes be a causal dependency between background updates. E.g. if update A adds a new column and then update B adds an index to that column, B must happen after A.

Synapse's background update machinery ensures(?) that updates with smaller ordering happen before updates with larger ordering. (I'm guessing that somewhere we `SELECT * FROM background_updates ORDER BY ordering ASC LIMIT 1;` or similar to pick the upgrade to run next.)

As for the value: `7324` is the concatenation of `73` and `24`, taken from the file path; `73` is the schema compat version and `24` is the migration version within schema compat `73.`

Reply:
Thank you!

Reply:
Note you can also add an explicit dependency via the `depends_on` column

Reply:
https://github.com/matrix-org/synapse/blob/373c485d8c7f39206bac60c6ef313b4a1978bbc0/synapse/storage/background_updates.py#L375-L411 has the machinery, I think